pkg/client/client.go

```diff
@@ -625,8 +628,24 @@ func (c *Client) CreateOrUpdatePrometheus(ctx context.Context, p *monv1.Promethe
 
 	required := p.DeepCopy()
 	mergeMetadata(&required.ObjectMeta, existing.ObjectMeta)
+	annotations := required.ObjectMeta.Annotations
+	labels := required.ObjectMeta.Labels
+	required.ObjectMeta = existing.ObjectMeta
+	required.ObjectMeta.Annotations = annotations
+	required.ObjectMeta.Labels = labels
+	required.Status = existing.Status
+
+	if reflect.DeepEqual(required, existing) {
+		// Nothing to do, as the currently existing resource is equivalent to the one that would be applied.
+		return nil
+	}
```

Comment:
This seems like a welcome change for all such instances in all components, to avoid `POST` requests to the APIServer for `Update`-ing only on a `!DeepEqual`. The controller-runtime model would have allowed us to serve the still-happening `GET` requests from the cache, but since that's out of the question, I believe this would still take some load off of the APIServer.